package/MDAnalysis/analysis/gnm.py

```diff
@@ -337,12 +353,9 @@ def generate_kirchoff(self):
             for jcounter in neighbour_atoms:
                 if jcounter > icounter and _dsq(positions[icounter], positions[jcounter]) <= cutoffsq:
                     iresidue, jresidue = residue_index_map[icounter], residue_index_map[jcounter]
-                    if self.MassWeight:
-                        contact = 1.0 / (len(self.ca.residues[iresidue].atoms) * len(self.ca.residues[jresidue].atoms)) ** 0.5
```

Comment:
Er, actually, *where* did you see another instance of this problem? `GNMAnalysis.generate_kirchoff()` does not use custom weights. I am not sure where else this needs fixing.